### PR TITLE
Models/gpor

### PR DESCRIPTION
**Pull request by tunczyk101**
- [Simple OR](https://dl.acm.org/doi/10.1007/3-540-44795-4_13)
- [GPOR](https://gpflow.github.io/GPflow/develop/notebooks/advanced/ordinal_regression.html)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -18,6 +18,8 @@ scikit-learn = "1.3.*"
 orf = "0.2.*"
 mizani = "0.9.*"  # orf dependencies
 mord = "0.7.*"
+#gpflow = "2.9.*"  # TODO: Package 'tensorflow-macos' is listed as a dependency of itself.
+# for Simple approach to OR: !git clone https://github.com/mosh98/Ordinal_Classifier.git
```
Comment (j-adamczyk) [inline pyproject.toml:22]: Just copy-paste that code

File: training/create_model.py [hunk 2]
```diff
@@ -189,6 +191,21 @@ def create_model(classifier_name: str):
             model = create_linear_ordinal_model("probit")
         case "linear_ordinal_model_logit":
             model = create_linear_ordinal_model("logit")
+        case "simple_or":
+            hyper_params = {
+                "max_features": [0.3],
+                "n_estimators": [100, 200, 500],
+                "criterion": ["gini", "entropy", "log_loss"],
```
Comment (j-adamczyk) [inline training/create_model.py:198]: entropy and log-loss are the same

File: training/create_model.py [hunk 3]
```diff
@@ -189,6 +191,21 @@ def create_model(classifier_name: str):
             model = create_linear_ordinal_model("probit")
         case "linear_ordinal_model_logit":
             model = create_linear_ordinal_model("logit")
+        case "simple_or":
+            hyper_params = {
+                "max_features": [0.3],
```
Comment (j-adamczyk) [inline training/create_model.py:196]: definitely check both sqrt and 0.3

File: training/models/gpor.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator, ClassifierMixin
+
+
+class GPOR(BaseEstimator, ClassifierMixin):
+    def __init__(self):
+        self.cutpoints = np.linspace(-10, 10, 22)
+        self.likelihood = gpflow.likelihoods.Ordinal(self.cutpoints)
+        self.kernel = gpflow.kernels.SquaredExponential()
```
Comment (j-adamczyk) [inline training/models/gpor.py:11]: There is also a `RBF` alias in GPFlow, it's more common name, so I would suggest using it

Comment (j-adamczyk) [reply training/models/gpor.py:11]: Maybe it would be useful to consider also a few other kernels, just to make sure, e.g. ArcCosine, Linear, Polynomial, RationalQuadratic

File: training/models/gpor.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+    def fit(self, X, y):
+        model = gpflow.models.VGP(
+            (X, pd.DataFrame(y)), kernel=self.kernel, likelihood=self.likelihood
+        )
+        optimizer = gpflow.optimizers.Scipy()
+        optimizer.minimize(
+            model.training_loss, model.trainable_variables, options=dict(maxiter=100)
```
Comment (j-adamczyk) [inline training/models/gpor.py:19]: Increasing `maxiter` improves performance, or not?

File: training/models/gpor.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+
+class GPOR(BaseEstimator, ClassifierMixin):
+    def __init__(self):
+        self.cutpoints = np.linspace(-10, 10, 22)
+        self.likelihood = gpflow.likelihoods.Ordinal(self.cutpoints)
+        self.kernel = gpflow.kernels.SquaredExponential()
+
+    def fit(self, X, y):
+        model = gpflow.models.VGP(
```
Comment (j-adamczyk) [inline training/models/gpor.py:14]: Maybe exact `GPR` will be more accurate?

File: training/models/simple_ordinal_classification.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
+from sklearn.base import BaseEstimator, ClassifierMixin
+from sklearn.ensemble import RandomForestClassifier
+
+from training.constants import RANDOM_STATE
+from training.models.Ordinal_Classifier.Ordinal_Classifier import OrdinalClassifier
+
+
+class SimpleOrdinalClassification(BaseEstimator, ClassifierMixin):
+    def __init__(self, max_features=0.3, n_estimators=100, criterion="gini"):
```
Comment (j-adamczyk) [inline training/models/simple_ordinal_classification.py:9]: You can use **kwargs and then get arbitrary parameters for RF here

Comment (j-adamczyk) [reply training/models/simple_ordinal_classification.py:9]: It will require using `setattr`

**Review by j-adamczyk — CHANGES_REQUESTED** [review]
